osquery/tables/system/darwin/running_apps.mm

```diff
@@ -0,0 +1,70 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#import <AppKit/AppKit.h>
+#import <Foundation/Foundation.h>
+
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genRunningApps(QueryContext& context) {
+  QueryData results;
+  if (!context.isAnyColumnUsed({"pid", "bundle_identifier", "is_active"})) {
+    return results;
+  }
+
+  NSWorkspace* workspace = [NSWorkspace sharedWorkspace];
+  NSRunningApplication* appInFocus = [workspace frontmostApplication];
```

Comment:
You should only need to call this if the query has `where is_active = 1`; for the other case you can use `NSRunningApplication isActive`. That'll let you reduce code duplication too (see comment later in this file).